src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -0,0 +1,211 @@
+        // to catch any exception that would be thrown and signal a test failure
+        Timer t = null;
+        AutoResetEvent are = new AutoResetEvent(false);
+        TimerCallback tc = new TimerCallback((object o) =>
+        {
+            t.Dispose();
+            are.Set();
+        });
+        t = new Timer(tc, null, 1, -1);
+        Assert.True(are.WaitOne(50));
+    }
+
+    [Fact]
+    public void Timer_CanBeDisposedMultipleTimes()
+    {
+        // There's nothing to validate besides that we don't throw an exception, so rely on xunit 
+        // to catch any exception that would be thrown and signal a test failure
+        TimerCallback tc = new TimerCallback((object o) => { });
+        var t = new Timer(tc, null, 100, -1);
+        for (int i = 0; i < 10; i++)
+            t.Dispose();
+    }
+
+    [Fact]
+    public void NonRepeatingTimer_ThatHasAlreadyFired_CanChangeAndFireAgain()
+    {
+        AutoResetEvent are = new AutoResetEvent(false);
+        TimerCallback tc = new TimerCallback((object o) => are.Set());
+        using (var t = new Timer(tc, null, 1, Timeout.Infinite))
+        {
+            Assert.True(are.WaitOne(50), "Should have received first timer event");
+            Assert.False(are.WaitOne(20), "Should not have received a second timer event");
+            t.Change(10, Timeout.Infinite);
+            Assert.True(are.WaitOne(50), "Should have received a second timer event after changing it");
+        }
+    }
+
+    [Fact, OuterLoop]
+    public void Running_Timer_CanBeFinalizedAndStopsFiring()
+    {
+        AutoResetEvent are = new AutoResetEvent(false);
+        TimerCallback tc = new TimerCallback((object o) => are.Set());
+        var t = new Timer(tc, null, 100, 500);
+        Assert.True(are.WaitOne(250), "Failed to get first timer fire");
+        t = null; // Remove our refence so the timer can be GC'd 
+        GC.Collect();
+        GC.WaitForPendingFinalizers();
+        GC.Collect();
+        Assert.False(are.WaitOne(750), "Should not have received a timer fire after it was collected");
+    }
+
+    [Fact, OuterLoop]
+    public void MultpleTimers_PeriodicTimerIsntBlockedByABlockedTimer()
+    {
+        object t1State = new object();
+        object t2State = new object();
+        AutoResetEvent areT1 = new AutoResetEvent(false);
+        AutoResetEvent areT2 = new AutoResetEvent(false);
+        TimerCallback tc = new TimerCallback((object o) =>
+        {
+            if (o == t1State)
+            {
+                areT2.WaitOne(250);
+                areT1.Set();
+            }
+            else
+            {
+                areT2.Set();
+            }
+        });
+
+        using (var t1 = new Timer(tc, t1State, 50, -1))
+        using (var t2 = new Timer(tc, t2State, 100, 100))
+        {
+            Assert.True(areT1.WaitOne(500), "Blocking callback prevented a second timer from firing");
+        }
+    }
+
+    [Fact, OuterLoop]
```

Comment:
Does this ManyTimers test take a long time?  We might be able to get away with it as inner loop.